### PR TITLE
ENH: Add ``--rescale-intensities`` and name_source to N4BiasFieldCorrection

### PR DESCRIPTION
**Pull request by oesteban**
## Summary

Fixes # .

## Acknowledgment

- [x] \(Mandatory\) I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

**Comment by effigies** [comment]
I wonder if it makes sense to deprecate `save_bias`, and just always output the bias image, using `name_template` (unless someone specifically undefines it). Presumably it's already calculated, and so the cost is minimal, and it would save the contortions of `_out_bias_file`.

Failing that, I would probably make `_out_bias_file` a property that reconstructs the file as called for or a hidden attribute that's `None` by default. Either of which will avoid the explicit `getattr`, `setattr`, `delattr` calls, which would make it more readable, IMO.

**Comment by oesteban** [comment]
> I wonder if it makes sense to deprecate `save_bias`, and just always output the bias image, using `name_template` (unless someone specifically undefines it). Presumably it's already calculated, and so the cost is minimal, and it would save the contortions of `_out_bias_file`.

As much as I like this idea, I think the `name_source` mechanism does not fully cover this behavior. In cases like this, it would be fine just generating the file anyways. For the general case, it would be best to allow some kind of metadata (`name_enable`) pointing to some other `traits.Bool` (in this case `save_bias`) that is used to check whether the name substitution should be executed.


> Failing that, I would probably make `_out_bias_file` a property that reconstructs the file as called for or a hidden attribute that's `None` by default. Either of which will avoid the explicit `getattr`, `setattr`, `delattr` calls, which would make it more readable, IMO.

Agreed, done.

**Comment by effigies** [comment]
Have you tested? I'm not positive that you'll get an absolute path as the output.

**Comment by oesteban** [comment]
> Have you tested? I'm not positive that you'll get an absolute path as the output.

0b695ae ensures the absolute path of the bias output and I have tested locally.

File: nipype/interfaces/ants/segmentation.py [hunk 1]
```diff
@@ -306,6 +308,15 @@ class N4BiasFieldCorrectionInputSpec(ANTSCommandInputSpec):
         usedefault=True,
         desc='copy headers of the original image into the '
         'output (corrected) file')
+    rescale_intensities = traits.Bool(
+        False, usedefault=True, argstr='-r',
```
Comment (effigies) [inline nipype/interfaces/ants/segmentation.py]: min_ver?

File: nipype/interfaces/ants/segmentation.py [hunk 2]
```diff
@@ -418,34 +417,35 @@ def _format_arg(self, name, trait_spec, value):
...
-            skip = []
-        skip += ['save_bias', 'bias_image']
+        skip = (skip or []) + ['save_bias', 'bias_image']
+        self._out_bias_file = None
+        if self.inputs.save_bias or isdefined(self.inputs.bias_image):
+            bias_image = self.inputs.bias_image
+            if not isdefined(bias_image):
+                bias_image = fname_presuffix(os.path.basename(self.inputs.input_image),
+                                             suffix='_bias', newpath=os.getcwd())
+            self._out_bias_file = bias_image
```
Comment (effigies) [inline nipype/interfaces/ants/segmentation.py]: I'd put the abspath here, rather than in fname_presuffix, so the bias_image case is covered as well.

**Comment by oesteban** [comment]
I just realized that this overlaps with #2996 on the ``--rescale-intensities`` parameter.

**Comment by rciric** [comment]
I closed mine for now -- I'll open a new one with just the other N4 parameters so that it doesn't conflict.

**Pull request merged**